app/coffee/RealTimeRedisManager.coffee

```diff
@@ -36,4 +36,9 @@ module.exports = RealTimeRedisManager =
 		# create a unique message id using a counter
 		message_id = "doc:#{HOST}:#{RND}-#{COUNT++}"
 		data?._id = message_id
-		pubsubClient.publish "applied-ops", JSON.stringify(data)
+		# publish on separate channels for individual projects and docs when
+		# configured (needs realtime to be configured for this too).
+		if Settings.publishOnIndividualChannels and data.doc_id?
```

Comment:
Mainly for my own edification... are there any cases where we'd expect `data.doc_id` to be missing here? If we're applying ops, it seems like we should always have a doc. It still seems worth checking but could be an error, if it should always be present.

Reply:
Yes, it should always present in docupdater, so the test for `data.doc_id?` is redundant. I'll remove it.